mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/DependencyResolver.java

```diff
@@ -30,21 +32,41 @@ private void addStrategies(Model model) {
         final StrategyDependencyResolver strategyDependencyResolver = new StrategyDependencyResolver();
         model.addDependency(smartTestingProviderDependency());
         final Map<String, Dependency> dependencies = strategyDependencyResolver.resolveDependencies();
-        final Map<String, String> strategyMismatch = new HashMap<>();
-        for (final String definedStrategy : strategies) {
+        final List<String> strategyMismatch = new ArrayList<>();
+        final List<String> registeredStrategies = new ArrayList<>();
+        for (int i=0; i < strategies.length; i++) {
+            String definedStrategy = strategies[i];
             if (!dependencies.containsKey(definedStrategy)) {
                 final String closestMatch = stringSimilarityCalculator.findClosestMatch(definedStrategy, dependencies.keySet());
-                strategyMismatch.put(definedStrategy, closestMatch);
+                if (configuration.isAutocorrect()) {
+                    if (registeredStrategies.contains(closestMatch)) {
+                        strategyMismatch.add(String.format("Autocorrected [%s] strategy to [%s] but it was already registered", closestMatch, definedStrategy));
+                    } else {
+                        final Dependency dependency = dependencies.get(closestMatch);
+                        model.addDependency(dependency);
+                        strategies[i]= closestMatch;
+                        registeredStrategies.add(closestMatch);
+                    }
+                } else {
+                    strategyMismatch.add(String.format("Unable to find strategy [%s]. Did you mean [%s]?", definedStrategy, closestMatch));
+                }
             } else {
-                final Dependency dependency = dependencies.get(definedStrategy);
-                model.addDependency(dependency);
+
+                if (registeredStrategies.contains(definedStrategy)) {
+                    strategyMismatch.add(String.format("Strategy [%s] was already registered or autocorrected", definedStrategy));
+                } else {
+                    final Dependency dependency = dependencies.get(definedStrategy);
+                    model.addDependency(dependency);
+                    registeredStrategies.add(definedStrategy);
+                }
             }
         }
-        strategyMismatch.forEach((selection, match) -> LOGGER.error("Unable to find strategy [%s]. Did you mean [%s]?", selection, match));
+        strategyMismatch.forEach(msg -> LOGGER.error(msg));
```

Comment:
* `Strategy [%s] was already registered or autocorrected`
 * `Autocorrected [%s] strategy to [%s] but it was already registered` 
this msgs belongs to `info` or `error`?

Reply:
It is an error.